src/main/java/frc/robot/subsystems/swerve/Swerve.java

```diff
@@ -283,8 +288,13 @@ protected void driveByState(ChassisSpeeds speeds, SwerveState swerveState) {
 		speeds = SwerveMath.factorSpeeds(speeds, swerveState.getDriveSpeed());
 		speeds = SwerveMath.applyDeadband(speeds);
 		speeds = getDriveModeRelativeSpeeds(speeds, swerveState);
-		speeds = SwerveMath.discretize(speeds);
-
+		double fudgeFactor = discritizationHelper.getInterpolatedValue(
```

Comment:
correct me if I'm wrong -- but this line causes you to run `createBoundingBox(...)` every call -- even though the bounding box remains the same (assuming no change in the knownPoints in runtime). This looks costly, in terms of compute power.